### PR TITLE
10843 construct config from v2 yaml

### LINKED ISSUE #10843: Convert v2 API test fragments to v3
While all programatically constructed test configs will be v3 (via API boosting rewriting), there are still v2 JSON/YAML fragments in test code. Many of these will "just work" as v3, but some may have v2 features that will have them treated as v2. This becomes confusing when they are extended by developers who want to model v3 features.

These need to be audited and updated as appropriate. Also, the names of methods such as `parseAccessLogFromV2Yaml` should be updated to reflect v3 when operating on fragments that should be v3.

This is somewhat similar to the work done by @dmitri-d for examples in https://github.com/envoyproxy/envoy/pull/10644 as part of https://github.com/envoyproxy/envoy/issues/9735.

CC @alyssawilk

### PR DESCRIPTION
**Pull request by ankatare**
Commit Message: changes related to constructconfigfromv2yaml
Additional Description:
Risk Level: NA
Testing:unit , integration and format testing
Docs Changes: No
Release Notes:
[Optional Runtime guard:]
Optional Fixes #10843 
[Optional Deprecated:]

### CODE REVIEW

**Comment by htuch** [comment]
@ankatare this overlaps with https://github.com/envoyproxy/envoy/pull/11894, can you make them non-overlapping or coalesce? Also, what does "10843 construct" mean in the title?

**Comment by ankatare** [comment]
@htuch  in this PR i modified for functions "constructConfigFromV2Yaml" wherever applicable in test cases. so it is separate from others as mentioned in PR #11894

**Comment by ankatare** [comment]
moreover there are unexpected build issues in this PR as well.

File: test/common/upstream/eds_speed_test.cc [hunk 1]
```diff
@@ -47,7 +47,7 @@ class EdsSpeedTest {
 
   void resetCluster(const std::string& yaml_config, Cluster::InitializePhase initialize_phase) {
     local_info_.node_.mutable_locality()->set_zone("us-east-1a");
-    eds_cluster_ = parseClusterFromV2Yaml(yaml_config);
+    eds_cluster_ = parseClusterFromV3Yaml(yaml_config);
```
Comment (htuch) [inline test/common/upstream/eds_speed_test.cc:50]: This PR has `parseClusterFromV3Yaml`, so does not seem orthogonal to https://github.com/envoyproxy/envoy/pull/11894. It would be helpful if you can split the PRs along some easily understandable line, or stage them one after the other.

Also, friendly ping on updating title/description, it's hard to figure out what "10843 construct" means.

Comment (ankatare) [reply test/common/upstream/eds_speed_test.cc:50]: @htuch i am not receiving any conflicts message for this PR. please suggest how to proceed.

Comment (ankatare) [reply test/common/upstream/eds_speed_test.cc:50]: @htuch seems some env issues in my local setup. i  recreated duplicate PR for this PR. please review it. #12072. 
           we can close this now.
